ext/dba/tests/setup/setup_dba_tests.inc

```diff
@@ -102,14 +102,28 @@ function run_standard_tests_ex(string $handler, string $name, LockFlag $lock, bo
     echo 'Try to remove key 1 again', \PHP_EOL;
     var_dump(dba_delete("key1", $db_writer));
 
-    // Fetch data
+    // Fetch and sort data. We sort to guarantee that the output is
+    // always the same across architectures. GDBM in particular does
+    // not guarantee that the insert order is the same as the
+    // iteration order with firstkey() and nextkey(). There is a
+    // Github report (issue 14786) of this actually happening. GDBM's
+    // own test suite sorts its output, suggesting that this is a
+    // reasonable workaround for the issue.
+    $output = [];
```

Comment:
If this may only affect GDBM, maybe we should sort only for that `$handler`? After all, having the fetch order the same as the insertion order may be regarded as a feature of other handlers.

Reply:
In the second commit you can see that there were already some other engines that had different orders. Those orders may still have been _consistent_, but who knows. I can't personally reproduce even the gdbm issue.

FWIW The TokyoCabinet, IBM dbm, and OpenBSD NDBM docs explicitly say the order is arbitrary. LMDB and QDBM don't seem to say either way.

Reply:
Amending the comment would be a good idea then